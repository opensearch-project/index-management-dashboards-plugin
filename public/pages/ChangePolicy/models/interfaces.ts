/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { Policy } from "../../../../models/interfaces";

export interface PolicyOption {
  label: string;
  value: Policy;
}
