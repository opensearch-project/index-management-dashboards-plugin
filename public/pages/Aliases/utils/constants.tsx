/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import { SortDirection } from "../../../utils/constants";
import { IAlias } from "../interface";

export const DEFAULT_PAGE_SIZE_OPTIONS = [5, 10, 20, 50];

export const DEFAULT_QUERY_PARAMS = {
  from: "0",
  size: "20",
  search: "",
  sortField: "alias" as keyof IAlias,
  sortDirection: SortDirection.DESC,
  status: "",
  dataSourceId: "",
};
