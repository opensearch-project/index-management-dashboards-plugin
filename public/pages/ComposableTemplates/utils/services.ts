/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { CommonService } from "../../../services";
import { TemplateItemRemote } from "../../../../models/interfaces";

export const getAllUsedComponents = async ({ commonService }: { commonService: CommonService }) => {
  const allTemplatesResponse = await commonService.apiCaller<{
    index_templates?: Array<{
      name: string;
      index_template: TemplateItemRemote;
    }>;
  }>({
    endpoint: "transport.request",
    data: {
      method: "GET",
      path: "/_index_template/*",
    },
  });

  if (allTemplatesResponse && allTemplatesResponse.ok) {
    return allTemplatesResponse.response.index_templates || [];
  }

  return [];
};
