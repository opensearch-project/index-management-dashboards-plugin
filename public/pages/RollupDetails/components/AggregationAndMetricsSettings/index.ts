/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import AggregationAndMetricsSettings from "./containers/AggregationAndMetricsSettings";

export default AggregationAndMetricsSettings;
