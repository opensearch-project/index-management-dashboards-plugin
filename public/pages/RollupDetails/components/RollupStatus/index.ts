/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import RollupStatus from "./RollupStatus";

export default RollupStatus;
