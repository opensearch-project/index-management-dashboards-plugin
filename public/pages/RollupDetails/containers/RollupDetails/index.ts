/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import RollupDetails from "./RollupDetails";

export default RollupDetails;
