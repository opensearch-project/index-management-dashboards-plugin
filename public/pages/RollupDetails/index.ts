/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import RollupDetails from "./containers/RollupDetails";

export default RollupDetails;
