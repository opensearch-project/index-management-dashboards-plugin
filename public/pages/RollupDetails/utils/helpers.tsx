/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { EuiFlexGroup, EuiFlexItem, EuiIcon, EuiText } from "@elastic/eui";
import { RollupMetadata } from "../../../../models/interfaces";

export const renderStatus = (metadata: RollupMetadata | undefined): JSX.Element => {
  if (metadata == null || metadata === undefined || metadata.rollup_metadata == null) return <dd>-</dd>;
  let icon;
  let iconColor;
  let textColor: "default" | "subdued" | "secondary" | "ghost" | "accent" | "warning" | "danger" | undefined;
  let text;
  switch (metadata.rollup_metadata.status) {
    case "failed":
      icon = "alert";
      iconColor = "danger";
      textColor = "danger";
      text = "Failed: " + metadata.rollup_metadata.failure_reason;
      break;
    case "finished":
      icon = "check";
      iconColor = "success";
      textColor = "secondary";
      text = "Complete";
      break;
    case "init":
      return (
        <EuiFlexGroup gutterSize="xs">
          <EuiFlexItem grow={false}>
            <EuiIcon size="s" type="clock" color="primary" />
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <EuiText size="xs" style={{ color: "#006BB4" }}>
              Initializing
            </EuiText>
          </EuiFlexItem>
        </EuiFlexGroup>
      );
    case "started":
      icon = "play";
      iconColor = "success";
      textColor = "secondary";
      text = "Started";
      break;
    case "stopped":
      icon = "stop";
      iconColor = "subdued";
      textColor = "subdued";
      text = "Stopped";
      break;
    default:
      return <dd>-</dd>;
  }

  return (
    <EuiFlexGroup gutterSize="xs">
      <EuiFlexItem grow={false}>
        <EuiIcon size="s" type={icon} color={iconColor} />
      </EuiFlexItem>
      <EuiFlexItem grow={false}>
        <EuiText size="xs" color={textColor}>
          {text}
        </EuiText>
      </EuiFlexItem>
    </EuiFlexGroup>
  );
};
