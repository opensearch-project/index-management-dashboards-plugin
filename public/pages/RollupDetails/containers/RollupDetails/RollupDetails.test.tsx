/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MemoryRouter as Router } from "react-router-dom";
import { Redirect, Route, RouteComponentProps, Switch } from "react-router-dom";
import { browserServicesMock, coreServicesMock } from "../../../../../test/mocks";
import { BrowserServices } from "../../../../models/interfaces";
import { ModalProvider, ModalRoot } from "../../../../components/Modal";
import { BREADCRUMBS, ROUTES } from "../../../../utils/constants";
import RollupDetails from "./RollupDetails";
import { ServicesConsumer, ServicesContext } from "../../../../services";
import { testRollup, testRollup2 } from "../../../../../test/constants";
import { CoreServicesContext } from "../../../../components/core_services";

function renderRollupDetailsWithRouter(initialEntries = ["/"]) {
  return {
    ...render(
      <Router initialEntries={initialEntries}>
        <CoreServicesContext.Provider value={coreServicesMock}>
          <ServicesContext.Provider value={browserServicesMock}>
            <ServicesConsumer>
              {(services: BrowserServices | null) =>
                services && (
                  <ModalProvider>
                    <ModalRoot services={services} />
                    <Switch>
                      <Route
                        path={ROUTES.ROLLUP_DETAILS}
                        render={(props: RouteComponentProps) => <RollupDetails {...props} rollupService={services.rollupService} />}
                      />
                      <Route path={ROUTES.EDIT_ROLLUP} render={(props) => <div>Testing edit rollup: {props.location.search}</div>} />
                      <Route path={ROUTES.ROLLUPS} render={(props) => <div>Testing rollup landing page</div>} />
                      <Redirect from="/" to={ROUTES.ROLLUP_DETAILS} />
                    </Switch>
                  </ModalProvider>
                )
              }
            </ServicesConsumer>
          </ServicesContext.Provider>
        </CoreServicesContext.Provider>
      </Router>
    ),
  };
}

describe("<RollupDetails /> spec", () => {
  it("renders the component", async () => {
    browserServicesMock.rollupService.getRollup = jest.fn().mockResolvedValue({
      ok: true,
      response: testRollup,
    });
    const { container } = renderRollupDetailsWithRouter();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("sets breadcrumbs when mounting", async () => {
    browserServicesMock.rollupService.getRollup = jest.fn().mockResolvedValue({
      ok: true,
      response: testRollup,
    });
    renderRollupDetailsWithRouter([`${ROUTES.ROLLUP_DETAILS}?id=${testRollup._id}`]);

    expect(coreServicesMock.chrome.setBreadcrumbs).toHaveBeenCalledTimes(2);
    expect(coreServicesMock.chrome.setBreadcrumbs).toHaveBeenCalledWith([
      BREADCRUMBS.INDEX_MANAGEMENT,
      BREADCRUMBS.ROLLUPS,
      { text: testRollup._id },
    ]);
  });

  it("can disable rollup job", async () => {
    browserServicesMock.rollupService.getRollup = jest.fn().mockResolvedValue({
      ok: true,
      response: testRollup,
    });

    browserServicesMock.rollupService.stopRollup = jest.fn().mockResolvedValue({
      ok: true,
      response: true,
    });
    const { getByTestId } = renderRollupDetailsWithRouter([`${ROUTES.ROLLUP_DETAILS}?id=${testRollup._id}`]);

    await waitFor(() => {});

    expect(getByTestId("disableButton")).toBeEnabled();

    expect(getByTestId("enableButton")).toBeDisabled();

    userEvent.click(getByTestId("disableButton"));

    await waitFor(() => {});

    expect(browserServicesMock.rollupService.stopRollup).toHaveBeenCalledTimes(1);
    expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledTimes(1);
  });

  it("shows toast when failed to disable rollup job", async () => {
    browserServicesMock.rollupService.getRollup = jest.fn().mockResolvedValue({
      ok: true,
      response: testRollup,
    });

    browserServicesMock.rollupService.stopRollup = jest.fn().mockResolvedValue({
      ok: false,
      response: "some error",
    });
    const { getByTestId } = renderRollupDetailsWithRouter([`${ROUTES.ROLLUP_DETAILS}?id=${testRollup._id}`]);

    await waitFor(() => {});

    expect(getByTestId("disableButton")).toBeEnabled();

    expect(getByTestId("enableButton")).toBeDisabled();

    userEvent.click(getByTestId("disableButton"));

    await waitFor(() => {});

    expect(browserServicesMock.rollupService.stopRollup).toHaveBeenCalledTimes(1);
    expect(coreServicesMock.notifications.toasts.addDanger).toHaveBeenCalledTimes(1);
  });

  it("can enable rollup job", async () => {
    browserServicesMock.rollupService.getRollup = jest.fn().mockResolvedValue({
      ok: true,
      response: testRollup2,
    });

    browserServicesMock.rollupService.startRollup = jest.fn().mockResolvedValue({
      ok: true,
      response: true,
    });

    const { getByTestId } = renderRollupDetailsWithRouter([`${ROUTES.ROLLUP_DETAILS}?id=${testRollup2._id}`]);

    await waitFor(() => {});

    expect(getByTestId("enableButton")).toBeEnabled();

    expect(getByTestId("disableButton")).toBeDisabled();

    userEvent.click(getByTestId("enableButton"));

    await waitFor(() => {});

    expect(browserServicesMock.rollupService.startRollup).toHaveBeenCalledTimes(1);
    expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledTimes(1);
  });

  it("can delete a rollup job", async () => {
    const rollups = [testRollup];
    browserServicesMock.rollupService.getRollups = jest
      .fn()
      .mockResolvedValueOnce({ ok: true, response: { rollups, totalRollups: 1 } })
      .mockResolvedValueOnce({ ok: true, response: { rollups: [], totalRollups: 0 } });
    browserServicesMock.rollupService.deleteRollup = jest.fn().mockResolvedValue({ ok: true, response: true });
    const { getByTestId } = renderRollupDetailsWithRouter([`${ROUTES.ROLLUP_DETAILS}?id=${testRollup._id}`]);

    await waitFor(() => {
      getByTestId("deleteButton");
    });

    userEvent.click(getByTestId("deleteButton"));

    await waitFor(() => getByTestId("deleteTextField"));

    expect(getByTestId("confirmModalConfirmButton")).toBeDisabled();

    await userEvent.type(getByTestId("deleteTextField"), "delete");

    expect(getByTestId("confirmModalConfirmButton")).toBeEnabled();

    userEvent.click(getByTestId("confirmModalConfirmButton"));

    await waitFor(() => {});

    expect(browserServicesMock.rollupService.deleteRollup).toHaveBeenCalledTimes(1);
    expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledTimes(1);
  }, 10000);

  it("can show a started rollup job", async () => {
    const startedJob = testRollup;
    startedJob.metadata.test1.rollup_metadata.status = "started";

    browserServicesMock.rollupService.getRollup = jest.fn().mockResolvedValue({
      ok: true,
      response: testRollup,
    });

    const { queryByText } = renderRollupDetailsWithRouter([`${ROUTES.ROLLUP_DETAILS}?id=${testRollup._id}`]);

    await waitFor(() => {});

    expect(queryByText("Started")).not.toBeNull();
  });

  it("can show a stopped rollup job", async () => {
    const stoppedJob = testRollup;
    stoppedJob.metadata.test1.rollup_metadata.status = "stopped";

    browserServicesMock.rollupService.getRollup = jest.fn().mockResolvedValue({
      ok: true,
      response: testRollup,
    });

    const { queryByText } = renderRollupDetailsWithRouter([`${ROUTES.ROLLUP_DETAILS}?id=${testRollup._id}`]);

    await waitFor(() => {});

    expect(queryByText("Stopped")).not.toBeNull();
  });
});
