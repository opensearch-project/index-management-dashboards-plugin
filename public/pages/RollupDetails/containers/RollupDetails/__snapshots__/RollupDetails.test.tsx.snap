// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RollupDetails /> spec renders the component 1`] = `
<div
  style="padding: 5px 50px;"
>
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
    style="padding: 0px 10px;"
  >
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        class="euiText euiText--small"
      >
        <h1
          class="euiTitle euiTitle--medium"
        />
      </div>
    </div>
    <div
      class="euiFlexItem"
    >
      <div
        class="euiHealth euiHealth--textSizeS"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            EuiIconMock
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            Disabled
          </div>
        </div>
      </div>
    </div>
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <button
            class="euiButton euiButton--primary euiButton--small euiButton-isDisabled"
            data-test-subj="disableButton"
            disabled=""
            type="button"
          >
            <span
              class="euiButtonContent euiButton__content"
            >
              <span
                class="euiButton__text"
              >
                Disable
              </span>
            </span>
          </button>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <button
            class="euiButton euiButton--primary euiButton--small"
            data-test-subj="enableButton"
            type="button"
          >
            <span
              class="euiButtonContent euiButton__content"
            >
              <span
                class="euiButton__text"
              >
                Enable
              </span>
            </span>
          </button>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <button
            class="euiButton euiButton--primary euiButton--small"
            type="button"
          >
            <span
              class="euiButtonContent euiButton__content"
            >
              <span
                class="euiButton__text"
              >
                View JSON
              </span>
            </span>
          </button>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <button
            class="euiButton euiButton--danger euiButton--small"
            data-test-subj="deleteButton"
            type="button"
          >
            <span
              class="euiButtonContent euiButton__content"
            >
              <span
                class="euiButton__text"
              >
                Delete
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--directionRow euiFlexGroup--responsive"
    >
      <div
        class="euiFlexItem"
      >
        <div
          class="euiText euiText--small"
        >
          <h2>
            General information
          </h2>
        </div>
      </div>
      <div
        class="euiFlexItem euiFlexItem--flexGrowZero"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <button
              class="euiButton euiButton--primary euiButton--small"
              data-test-subj="EditButton"
              type="button"
            >
              <span
                class="euiButtonContent euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Edit
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div>
      <div
        class="euiFlexGrid euiFlexGrid--gutterLarge euiFlexGrid--fourths euiFlexGrid--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--small"
          >
            <dt>
              Name
            </dt>
            <dd />
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--small"
          >
            <dt>
              Source index
            </dt>
            <dd />
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--small"
          >
            <dt>
              Target index
            </dt>
            <dd />
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--small"
          >
            <dt>
              Schedule
            </dt>
            <dd />
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--small"
          >
            <dt>
              Description
            </dt>
            <dd>
              -
            </dd>
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--small"
          >
            <dt>
              Last updated
            </dt>
            <dd>
              -
            </dd>
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--small"
          >
            <dt>
              Pages per execution
            </dt>
            <dd>
              1000
            </dd>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--s"
      />
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--directionRow euiFlexGroup--responsive"
    >
      <div
        class="euiFlexItem"
      >
        <div
          class="euiText euiText--small"
        >
          <h2>
            Rollup status
          </h2>
        </div>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div>
      <div
        class="euiFlexGrid euiFlexGrid--gutterLarge euiFlexGrid--fourths euiFlexGrid--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--small"
          >
            <dt>
              Current rollup window
            </dt>
            <dd>
              -
            </dd>
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--small"
          >
            <dt>
              Status
            </dt>
            <dd>
              -
            </dd>
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--small"
          >
            <dt>
              Rollup indexed
            </dt>
            <dd>
              -
            </dd>
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--small"
          >
            <dt>
              Indexed time (ms)
            </dt>
            <dd>
              -
            </dd>
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--small"
          >
            <dt>
              Document processed
            </dt>
            <dd>
              -
            </dd>
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--small"
          >
            <dt>
              Search time (ms)
            </dt>
            <dd>
              -
            </dd>
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--small"
          >
            <dt>
              Page processed
            </dt>
            <dd>
              -
            </dd>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--s"
      />
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--directionRow euiFlexGroup--responsive"
    >
      <div
        class="euiFlexItem"
      >
        <div
          class="euiText euiText--small"
        >
          <h2>
            Aggregation and metrics settings
          </h2>
        </div>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div>
      <div
        class="euiText euiText--small"
      >
        <h3>
          Additional metrics
        </h3>
      </div>
      <div
        class="euiFlexGrid euiFlexGrid--gutterLarge euiFlexGrid--thirds euiFlexGrid--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--extraSmall"
          >
            <dt>
              Timestamp field
            </dt>
            <dd />
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--extraSmall"
          >
            <dt>
              Interval
            </dt>
            <dd />
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--extraSmall"
          >
            <dt>
              Timezone
            </dt>
            <dd>
              UTC +0
            </dd>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--s"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <div
            class="euiText euiText--small"
          >
            <h3>
              Additional aggregations
            </h3>
          </div>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <div
            class="euiText euiText--small"
          >
            <div
              class="euiTextAlign euiTextAlign--left"
            >
              <div
                class="euiTextColor euiTextColor--subdued"
              >
                <h3>
                  (0)
                </h3>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiText euiText--extraSmall"
      >
        <dd>
          No fields added for aggregation
        </dd>
      </div>
      <div
        class="euiSpacer euiSpacer--s"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <div
            class="euiText euiText--small"
          >
            <h3>
              Additional metrics
            </h3>
          </div>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <div
            class="euiText euiText--small"
          >
            <div
              class="euiTextAlign euiTextAlign--left"
            >
              <div
                class="euiTextColor euiTextColor--subdued"
              >
                <h3>
                  (0)
                </h3>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiText euiText--extraSmall"
      >
        <dd>
          No fields added for metrics
        </dd>
      </div>
      <div
        class="euiSpacer euiSpacer--s"
      />
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
</div>
`;
